src/control/lib/hostlist/hostlist.go

```diff
@@ -248,8 +254,9 @@ func parseBracketedHostList(input, rangeSep, rangeOp string) (*HostList, error)
 }
 
 // Create creates a new HostList from the supplied string representation.
-func Create(stringHosts string) (*HostList, error) {
-	return parseBracketedHostList(stringHosts, outerRangeSeparators, rangeOperator)
+func Create(stringHosts string, nameOptional bool) (*HostList, error) {
```

Comment:
embedding a NumericRange in hostRange doesn't sound like a tidier change than what I have currently (recent push), would you mind reviewing again please? it's not currently modifying the public API exposed in hostlist/api.go.

Reply:
It is still modifying the public (exported) API, though. This `Create()` function is part of that API. The functions in `api.go` are just convenience functions similar to the python API.

As it is now in this PR, the result of `hostlist.Create()` is something that may or may not actually be a valid hostlist (a `HostList` without hostnames is not valid). That is ambiguity that I do not want to introduce.

I don't mind adding some implementation complexity or code redundancy in order to keep the package unsurprising and easy to reason about. Context-dependent validity is not easy to reason about or test.

Instead of changing the signature of `Create()`, which should return a valid `HostList` or an error, you could move the bulk of this code into an unexported `create()` that is used behind the scenes, and then `Create()` and the other public functions just call that one.

The goal should be to add new functionality to the package without causing ripples of updates to code that didn't "need" to change.